resolve/index.go

```diff
@@ -61,13 +61,23 @@ type Resolver interface {
 	Resolve(c *config.Config, ix *RuleIndex, rc *repo.RemoteCache, r *rule.Rule, imports interface{}, from label.Label)
 }
 
+// CrossResolver is an interface that language extensions can implement to provide
+// custom dependency resolution logic for other languages.
+type CrossResolver interface {
+	// CrossResolve attempts to resolve an import string to a rule for languages
+	// other than the implementing extension. lang is the langauge of the rule
+	// with the dependency.
+	CrossResolve(c *config.Config, ix *RuleIndex, rc *repo.RemoteCache, imp ImportSpec, lang string) []FindResult
+}
+
 // RuleIndex is a table of rules in a workspace, indexed by label and by
 // import path. Used by Resolver to map import paths to labels.
 type RuleIndex struct {
 	rules     []*ruleRecord
 	labelMap  map[label.Label]*ruleRecord
 	importMap map[ImportSpec][]*ruleRecord
 	mrslv     func(r *rule.Rule, pkgRel string) Resolver
+	crslvs    []CrossResolver
```

Comment:
Rename to `crossResolvers`. Too much of an abbreviation I think. (`mrslv` is an unfortunate precedent).